### ISSUE TITLE
why-did-you-render causes React Devtools to crash on some components

### ISSUE THREAD

**Issue by jansedlon** [welldone-software/why-did-you-render#77]
This issue is being discussed here [React Devtools](https://github.com/facebook/react/issues/17670)

**Comment by justingrant** [comment]
I did some troubleshooting and it looks like one symptom of the problem is that hook state is incorrect-- for example, a useRef hook that was initialized with a function subsequently returns a ref object whose `current` contains a `Date` object, not a function. This `Date` instance looks suspiciously like the state from a different hook.

Note that this problem, unlike #78, doesn't rely on StrictMode. It will also happen in a non-StrictMode tree.

@jansedlon - do you have a repro you could post?  If not, I'll continue to work on finding a simple repro for this problem.

**Comment by justingrant** [comment]
OK, here's a 14-line repro: 
```js
import React, { useState, useMemo } from "react";
import ReactDOM from "react-dom";

const whyDidYouRender = require("@welldone-software/why-did-you-render");
whyDidYouRender(React);

function Repro() {
  const [count] = useState(0);
  const obj = useMemo(() => ({ count }), [count]);
  return <div>count: {obj.count}</div>;
}

const App = () => <Repro />;
ReactDOM.render(<App />, document.getElementById("root"));
```
  
**Repro Steps**

1. `git clone https://github.com/justingrant/repro-why-render-hooks` (this is a just-created CRA app with non-essential files removed and only one dependency added: `@welldone-sofwtare/why-did-you-render`)
2. `cd repro-why-render-hooks`
3. `npm i`
4. `npm start`
5. open http://localhost:3000 in Chrome
6. right-click + Inspect to open dev tools
7. select the ⚛Profiler tab
8. click the "Reload and start profiling" button (it looks like a little refresh icon)
9. click on the `Repro` component in the Flamegraph view

Expected: no crash
Actual: crash! `TypeError: Cannot read property 'push' of undefined` inside React Dev Tools

**Comment by vzaidman** [comment]
I'm on it.

**Comment by vzaidman** [comment]
@justingrant for me it crashes even if i remove WDYR.
can you check again please?

**Comment by jansedlon** [comment]
@vzaidman I know it's a silly question but did you cleared cache?

**Comment by vzaidman** [comment]
oops i removed it on "App" but there's an identical file called "index.js" with the same content LOL.
I keep on working on it.

**Comment by vzaidman** [comment]
fixed in 3.5.0.

i refactored the hooks to be less "patchy" for different reason just today and it seemed to help

**Comment by justingrant** [comment]
Oops, sorry @vzaidman for forgetting to delete that extra file. I started with a create-react-app template and then deleted a bunch of files... but I forgot to delete App.js after moving its content into index.js. Sorry for the confusion, and thanks for the quick fix!  I'll try 3.5.0 on my larger app and will let you know if I'm still running into crashes or hooks problems there.
